diff --git a/flow/platforms/asap7/config.mk b/flow/platforms/asap7/config.mk
index 2c9edd59..a0b3b46f 100644
--- a/flow/platforms/asap7/config.mk
+++ b/flow/platforms/asap7/config.mk
@@ -48,15 +48,15 @@ export WC_VOLTAGE          = 0.63
 # Dont use cells to ease congestion
 # Specify at least one filler cell if none
 export DONT_USE_CELLS          = *x1p*_ASAP7* *xp*_ASAP7*
-export DONT_USE_CELLS          += SDF* ICG* DFFH*
+export DONT_USE_CELLS          += SDF* ICG* DFFH* ASYNC_DFFH*
 
 # Yosys mapping files
 # Blackbox - list all standard cells and cells yosys should treat as blackboxes
-export BLACKBOX_V_FILE         = $(PLATFORM_DIR)/yoSys/asap7sc7p5t.blackbox.v
+# export BLACKBOX_V_FILE         = $(PLATFORM_DIR)/yoSys/asap7sc7p5t.blackbox.v
 export LATCH_MAP_FILE          = $(PLATFORM_DIR)/yoSys/cells_latch.v
 export CLKGATE_MAP_FILE        = $(PLATFORM_DIR)/yoSys/cells_clkgate.v
 export ADDER_MAP_FILE         ?= $(PLATFORM_DIR)/yoSys/cells_adders.v
-export BLACKBOX_MAP_TCL        = $(PLATFORM_DIR)/yoSys/blackbox_map.tcl
+# export BLACKBOX_MAP_TCL        = $(PLATFORM_DIR)/yoSys/blackbox_map.tcl
 
 # Set yosys-abc clock period to first "clk_period" value or "-period" value found in sdc file
 export ABC_CLOCK_PERIOD_IN_PS ?= $(shell sed -nr "s/^set\s+clk_period\s+(\S+).*|.*-period\s+(\S+).*/\1\2/p" $(SDC_FILE) | head -1 | awk '{print $$1}')
@@ -101,7 +101,7 @@ export TAPCELL_TCL             = $(PLATFORM_DIR)/openRoad/tapcell.tcl
 # TritonCTS options
 export CTS_BUF_CELL            ?= BUFx4_ASAP7_75t_R
 
-export CTS_BUF_DISTANCE        = 60
+# export CTS_BUF_DISTANCE        ?= 60
 
 # Fill cells used in fill cell insertion
 export FILL_CELLS              = "FILLERxp5_ASAP7_75t_R"
@@ -113,6 +113,9 @@ export SET_RC_TCL              = $(PLATFORM_DIR)/setRC.tcl
 export MIN_ROUTING_LAYER       = M2
 export MAX_ROUTING_LAYER       = M7
 
+# Define fastRoute tcl
+export FASTROUTE_TCL = $(PLATFORM_DIR)/fastroute.tcl
+
 # KLayout technology file
 export KLAYOUT_TECH_FILE       = $(PLATFORM_DIR)/KLayout/asap7.lyt
 
diff --git a/flow/scripts/cts.tcl b/flow/scripts/cts.tcl
index 87c672dc..614c8046 100644
--- a/flow/scripts/cts.tcl
+++ b/flow/scripts/cts.tcl
@@ -2,6 +2,8 @@ utl::set_metrics_stage "cts__{}"
 source $::env(SCRIPTS_DIR)/load.tcl
 load_design 3_place.odb 3_place.sdc "Starting CTS"
 
+set_dont_use $::env(DONT_USE_CELLS)
+
 # Clone clock tree inverters next to register loads
 # so cts does not try to buffer the inverted clocks.
 repair_clock_inverters
@@ -35,8 +37,6 @@ clock_tree_synthesis -root_buf "$::env(CTS_BUF_CELL)" -buf_list "$::env(CTS_BUF_
 
 set_propagated_clock [all_clocks]
 
-set_dont_use $::env(DONT_USE_CELLS)
-
 utl::push_metrics_stage "cts__{}__pre_repair"
 source $::env(SCRIPTS_DIR)/report_metrics.tcl
 
diff --git a/flow/scripts/floorplan.tcl b/flow/scripts/floorplan.tcl
index 5f588ab0..4b215a7c 100644
--- a/flow/scripts/floorplan.tcl
+++ b/flow/scripts/floorplan.tcl
@@ -54,6 +54,7 @@ remove_buffers
 
 ##### Restructure for timing #########
 if { [info exist ::env(RESYNTH_TIMING_RECOVER)] && $::env(RESYNTH_TIMING_RECOVER) == 1 } {
+  set_dont_use $::env(DONT_USE_CELLS)
   repair_design
   repair_timing
   # pre restructure area/timing report (ideal clocks)
diff --git a/flow/scripts/global_route.tcl b/flow/scripts/global_route.tcl
index 528f12f6..d4dabd67 100644
--- a/flow/scripts/global_route.tcl
+++ b/flow/scripts/global_route.tcl
@@ -15,7 +15,8 @@ if {[info exist env(FASTROUTE_TCL)]} {
 }
 
 global_route -guide_file $env(RESULTS_DIR)/route.guide \
-               -congestion_iterations 100 \
+               -congestion_iterations 500 \
+               -allow_congestion \
                -verbose
 
 set_propagated_clock [all_clocks]
