diff --git a/flow/platforms/nangate45/config.mk b/flow/platforms/nangate45/config.mk
index 7174863f..6ef97914 100644
--- a/flow/platforms/nangate45/config.mk
+++ b/flow/platforms/nangate45/config.mk
@@ -106,7 +106,7 @@ export CDL_FILE = $(PLATFORM_DIR)/cdl/NangateOpenCellLibrary.cdl
 export TEMPLATE_PGA_CFG ?= $(PLATFORM_DIR)/template_pga.cfg
 
 # OpenRCX extRules
-export RCX_RULES                           = ./platforms/$(PLATFORM)/rcx_patterns.rules
+export RCX_RULES               = $(PLATFORM_DIR)/rcx_patterns.rules
 # ---------------------------------------------------------
 #  IR Drop
 # ---------------------------------------------------------
